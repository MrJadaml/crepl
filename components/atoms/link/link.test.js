import React from 'react';
import Link from './index';
import { cleanup, render } from '@testing-library/react';
import { getByText } from '@testing-library/dom';

afterEach(cleanup);

describe('Link', () => {
  it('should render an anchor tag', () => {
    const { asFragment } = render(
      <Link href="#foo">
        Foo link
      </Link>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render children', () => {
    const { getByRole } = render(
      <Link href="#bar">
        Bar link
      </Link>
    );

    const link = getByRole('link');

    expect(link.textContent).toBe('Bar link');
  });
});
