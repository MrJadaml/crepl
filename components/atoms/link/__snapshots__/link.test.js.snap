// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render a Link 1`] = `
<DocumentFragment>
  <a
    class="link"
    href="#foo"
  >
    Foo link
  </a>
</DocumentFragment>
`;

exports[`Link should render a label 1`] = `
<DocumentFragment>
  <a
    class="link"
    href="#foo"
  >
    Foo label
  </a>
</DocumentFragment>
`;

exports[`Link should render an anchor tag 1`] = `
<DocumentFragment>
  <a
    class="link"
    href="#foo"
  >
    Foo link
  </a>
</DocumentFragment>
`;
